modules/affile/affile-source.c

```diff
@@ -456,6 +456,7 @@ affile_sd_new_instance(gchar *filename, GlobalConfig *cfg)
   self->super.super.super.deinit = affile_sd_deinit;
   self->super.super.super.notify = affile_sd_notify;
   self->super.super.super.free_fn = affile_sd_free;
+  self->super.super.super.generate_persist_id = affile_sd_format_persist_name;
```

Comment:
We'll need a wrapper around affile_sd_format_persist_name to accept const LogPipe* and turned that into the real "class". Now this code and the other similar give warning of incompatible type.